<launch>

    <include file="$(find habitat_ros)/launch/habitat_agent.launch">
        <arg name="agent_type" value="random_walk"/>
        <arg name="test_scene" value="/media/junting/SSD_data/habitat_data/scene_datasets/mp3d/v1/scans/17DRP5sb8fy/17DRP5sb8fy.glb"/>
    </include>

    <include file="$(find habitat_ros)/launch/rtabmap.launch">
        <arg name="rtabmap_args" value="--delete_db_on_start"/>
        <!-- <arg name="ground_truth_frame_id" value="world"/> -->
        <arg name="frame_id" value="base_link"/>
        <arg name="wait_for_transform" value="0.5"/>
        <arg name="rviz" value="false"/>
        <arg name="rtabmapviz" value="true"/>
        <arg name="approx_sync" value="false"/>
        <!-- <arg name="approx_rgbd_sync" value="false"/> -->
        
        <arg name="depth_topic" value="/camera/depth/image"/>
        <arg name="rgb_topic" value="/camera/rgb/image"/>
        <arg name="camera_info_topic" value="/camera/rgb/camera_info"/>

        <!-- custom arguments -->
        <arg name="grid_cellsize" value="0.015"/>
        <arg name="grid_depthdecimation" value="1"/> 
        <!-- <arg name="use_action_for_goal" value="false" /> -->
        <!-- <arg name="map_topic" default="grid_map" />
        <arg name="move_base_topic" default="move_base" /> -->

    </include>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find habitat_ros)/rviz/rgbd.rviz"/>

    <!-- run m-explore for frontier-based exploration -->
    <!-- <include file="$(find explore_lite)/launch/explore.launch"/> -->

</launch>